### ISSUE TITLE
Add support for PostgreSQL

### ISSUE THREAD

**Issue by pierre** [killbill/killbill#273]
Kill Bill already supports MySQL and H2, but a little more work is required for PostgreSQL. This is the master ticket to track the work involved:

* DDL files need to be updated, mainly to use `SERIAL` and/or a sequence instead of `auto_increment`. It would be nice to have a single set of files for all database types (for H2, we already trick it using the C-style comments, maybe we can do something similar for PostgreSQL).
* A PostgreSQL specific [locker](https://github.com/killbill/killbill-commons/tree/master/locker/src/main/java/org/killbill/commons/locker) needs to be implemented, using `pg_try_advisory_lock` and `pg_advisory_unlock` (note that PostgreSQL [doesn't take a string](http://www.postgresql.org/docs/9.1/static/functions-admin.html#FUNCTIONS-ADVISORY-LOCKS) as the lock key, so some deterministic hashing is needed).
* If PostgreSQL support is merged into the core, we need some regression tests, using an [embedded db](https://github.com/killbill/killbill-commons/tree/master/embeddeddb). The folks at airlift have published their wrapper on [GitHub](https://github.com/airlift/testing-postgresql-server). A corresponding build [profile](https://github.com/killbill/killbill-oss-parent/blob/master/pom.xml#L1599) would be nice.
* In killbill, `GlobalLockerModule` and `UtilTestSuiteWithEmbeddedDB` will need to be updated.

Nice to have:
* Generate the DDL file from ActiveRecord migrations in plugins (such as [Stripe](https://github.com/killbill/killbill-stripe-plugin/blob/master/db/ddl.sql))
* Update the [Analytics plugin](https://github.com/killbill/killbill-analytics-plugin) DDL and queries

**Comment by abeck87** [comment]
I was thinking about converting the DDL to postgres, but the Locker comment has me questioning if the app would be functional without it. What would be the risk if I proceed? Thanks.

**Comment by sbrossie** [comment]
Kill Bill needs the Locker to function properly; you have several choices:
1. If you work with only one instance-- unlikely in a production setting--
it is easy to implement a version of a Locker by just using regular locks
2. If not, then either you try to implement it using the primitives
mentioned in the ticket  (pg_try_advisory_lock and pg_advisory_unlock ), or
use a different scheme to implement the distributed lock, for e.g using
Zookeeper <https://zookeeper.apache.org/>)

**Comment by pierre** [comment]
This is blocked by https://github.com/killbill/killbill/issues/382.

**Comment by pierre** [comment]
Let's focus https://github.com/killbill/killbill/issues/382 on `TestConsumableInArrear` failures and keep this ticket for PostgreSQL support only.

Remaining work:

* Investigate @sbrossie's comment on our [queue code](https://github.com/killbill/killbill/issues/382#issuecomment-139060036).
* https://github.com/killbill/killbill-platform/pull/15

**Comment by pierre** [comment]
> Investigate @sbrossie's comment on our queue code.

We verified that the behavior of `claimEntry` is as expected.
